autosynth/synth.py

```diff
@@ -357,6 +370,26 @@ def synthesize_loop(
     if not toolbox.versions:
         return 0  # No versions, nothing to synthesize.
 
+    # Synthesize the library with the most recent versions of all sources.
+    youngest = len(toolbox.versions) - 1
+    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
+    if not has_changes:
+        if not toolbox.metadata_contains_generated_files(
+            toolbox.branch
+        ) and toolbox.metadata_contains_generated_files(toolbox.sub_branch(youngest)):
+            # Special case: the repo owner turned on obsolete file tracking.
+            # Generate a one-time PR containing only metadata changes.
+            executor.check_call(["git", "checkout", toolbox.branch])
```

Comment:
@SurferJeffAtGoogle if we turn this setting on globally, I gather we'll get a one time PR across all our repositories?